src/math/sqrt.rs

```diff
@@ -131,19 +131,19 @@ pub fn sqrt(x: f64) -> f64 {
         }
         m -= i - 1;
         ix0 |= (ix1 >> (32 - i) as usize).0 as i32;
-        ix1 = ix1 << i as usize;
+        ix1 <<= i as usize;
     }
     m -= 1023; /* unbias exponent */
-    ix0 = (ix0 & 0x000fffff) | 0x00100000;
+    ix0 = (ix0 & 0x_000f_ffff) | 0x_0010_0000;
     if (m & 1) == 1 {
         /* odd m, double x to make it even */
-        ix0 += ix0 + ((ix1 & sign) >> 31).0 as i32;
+        ix0 = ix0 + ix0 + ((ix1 & sign) >> 31).0 as i32;
```

Comment:
Why this change? This seems inconsistent with the change above (`ix1 = ix1 << i` to `ix1 <<= i`).

Reply:
I made this change because clippy warning.
I can revert and make clippy shut up.

Due to the fact that the variable is present on the right more than once, compiler can't optimize this code. Both variants are equivalent.
You can see ASM here.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=58c8753c3ee4954820234f64cd081db5

Reply:
Why does clippy warn in this case? I would expect the code to be equivalent, it is just a bit more readable to use `+=`.

For looking at ASM, I recommend [rust.godbolt.org](https://rust.godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(j:1,lang:rust,source:'pub+fn+inc(mut+a:+i32)+-%3E+i32+%7B%0A++++a+%2B%3D+1%3B%0A++++a%0A%7D%0A%0Apub+fn+foo(mut+a:+i32)+-%3E+i32+%7B%0A++++a+%2B%3D+a%2B1%3B%0A++++a%0A%7D%0A%0Apub+fn+bar(mut+a:+i32)+-%3E+i32+%7B%0A++++a+%3D+a%2Ba%2B1%3B%0A++++a%0A%7D'),l:'5',n:'0',o:'Rust+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:nightly,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'1',trim:'1'),lang:rust,libs:!(),options:'',source:1),l:'5',n:'0',o:'rustc+nightly+(Editor+%231,+Compiler+%231)+Rust',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4).

Reply:
Ok.

Reply:
The warning is because it looks like an improper refactor of `x = x + 1` into `x += x + 1` instead of `x += 1`.